Adds large cardboard boxes

Incoming5643 opened the pull request:
![341s6](https://cloud.githubusercontent.com/assets/4176358/9827337/3f6ff210-58af-11e5-9542-17b3c78db1a9.png)

Adds large cardboard boxes to the game, constructable from four sheets of cardboard. Roughly 20% of all generic closets on the map will now instead spawn as these.

Cardboard boxes are closets that:
- Can only carry a single person
- Are easy to destroy
- Are flammable
- Can't be welded shut

They can also can be used for more "tactical" purposes, at the same speed as walking.

Takes the opportunity to remove some object unfriendly coding.

Incoming5643 committed:
Adds large cardboard boxes to the game, constructable from five sheets of cardboard.

Cardboard boxes are closets that:
*Can only carry a single person
*Are easy to destroy
*Are flammable
*Can't be welded shut

They can also can be used for more "tactical" purposes, at the same speed as walking.

Takes the opportunity to remove some object unfriendly coding.

If this is accepted they'll be a pull a little later to replace a few closets on the maps with boxes (or maybe code to have generic closets on the map to occasionally spawn as boxes instead).

xxalpha commented:
Screenshots to see what it looks like?

Incoming5643 commented:
There's no sprites yet :see_no_evil:

Iamgoofball commented:
:+1:

Nitrousoxide commented:
Please make them look like the cardboard cutout of a lizard maid if you run with them on.

PKPenguin321 commented:
godlike :+1:

duncathan commented:
!

bgobandit commented:
if runtime is near a large cardboard box she should run inside it

Fox-McCloud commented:
>They can also can be used for more "tactical" purposes, at the same speed as walking.

Ha..

Aloraydrel commented:
!

Kromgar commented:
Now have cardboard boxes in maint and suddenly shadowlings hiding in cardboard boxes is new meta

ike709 commented:
Make it so you can attach a poster to it when standing up with it and walking around.

:+1:

Nienhaus commented:
Done https://www.dropbox.com/s/wjkvi28quub4kip/MGS.dmi?dl=0

peoplearestrange commented:
:+1: Nienhaus comes up trumps with a sprite again.

Dis gun be good.

Incoming5643 commented:
Excellent Nienhaus, age hasn't slowed you down one bit

Fox-McCloud commented:
There needs to be a subtype of this box that has a "!" sound that plays when you open it.

Nienhaus commented:
@Incoming5643 Later tonight I'm going to improve that. I spent like 10 minutes on it this morning when I woke up.

Incoming5643 committed:
Implemented Nienhaus' cardboard box sprite (pending apparently better sprite(s) coming)

Every roundstart generic locker now has a 20% chance of instead being a cardboard box (this is not the same thing as "20% of all lockers are now boxes" but it's a rough approximation).

As an easter egg, if you open a box containing a person everyone in sight will get a ! alert. This effect can only be triggered if someone other than the person in the box is the one who opens it. As a further safeguard against spamming this can happen at most once every minute.

Incoming5643 commented:
Alright well I implemented the old sprite anyways, but if you get something better drawn just post it here and I'll update it.

Also added the ! easter egg in this commit

Nienhaus commented:
@Incoming5643 https://www.dropbox.com/s/wjkvi28quub4kip/MGS.dmi?dl=0 Beep

Incoming5643 committed:
You are above even The Box. I hereby award you the title of Big Box.

Incoming5643 commented:
Done.

Also fairly obvious to anyone who looks at the the code but no, as far as I can tell we can't use THE sound for the alert easter egg, I can't find anything on its licensing, which probably means it was in house.

Nienhaus commented:
Yay!

Iamgoofball commented:
@Incoming5643 we can do a very close replica + idk i think parody act???

Shadowlight213 commented:
It should be partially invisible to sec officers at range

tigercat2000 commented on code/game/objects/structures/crates_lockers/closets.dm:
```diff
@@ -15,13 +15,20 @@
 
 /obj/structure/closet/New()
 	..()
+	if(type == /obj/structure/closet && ticker && ticker.current_state == GAME_STATE_PREGAME && prob(20)) //only roundstart generic closets
```
Couldn't this be put under closet/initialize()?

Incoming5643 replied:
Probably. I wasn't 100% sure if structures that were created during the initialization step would be properly initialized themselves, so I just stuck with New()

Pyko1 replied:
That seems to be a really shitty thing for anyone who maps. As you can no longer properly map a crate that will actually be a crate for sure.

Would be a lot better with some new item that has chance to be a crate or cardboard box, and then editing that to the map. Instead of forcing 20% of crates that are and ever will be to be a cardboarbox.

Incoming5643 replied:
It's only generic closets, the grey ones with nothing on them that are found almost exclusively in maintenance. All crates are still crates, and all special lockers are still those lockers.

I can do that later and revert this bit, it's just nothing screams "This will conflict for the next six weeks" quite like a map change that touches a lot of tiles

RemieRichards replied:
Put this in initialise or remove it and PR it later.
Merge chances missed: 1.

callanrocks commented:
They should make you unable to be tracked by the AI like agent cards yes/yes/yes?

Nienhaus commented:
@RemieRichards Merge pls

Miauw62 commented on code/game/objects/structures/crates_lockers/closets/cardboardbox.dm:
```diff
@@ -0,0 +1,59 @@
+				egg_cooldown = 1
+				spawn(600)
+					egg_cooldown = 0
+	..()
```
tbh im fine with people sneaking around in cardboard boxes but this is an """easter egg""" that triggers literally any time a box with a mob in it is opened, which is pretty shitty and will get obnoxious really fast.

RemieRichards replied:
it's actually only once per minute.
but yeah this is obnoxious.

Fox-McCloud replied:
Perhaps it could be a subtype that does it?

Incoming5643 replied:
It only happens when the person who opens the box ISN'T the person in the box. But yeah originally it was going to opperate on a low probability chance or only be able to activate once per box.

What would be more palatable to people?

duncathan replied:
if(prob(TCMB*0.5)) for maximum unreadability :^)

Iamgoofball replied:
make it once every 5 minutes?

Incoming5643 committed:
Upgrades the anti spam flag to five minutes

Come in from downwind, then pull up fast! Get ready to fast rope down to Shell One! Five minutes to ETA! Alpha, your top priority is to rescue and safeguard the President! Team Bravo, watch out for Stillman's back, and get those C4's disarmed!

Dorsisdwarf commented on code/game/objects/structures/crates_lockers/closets/cardboardbox.dm:
```diff
@@ -0,0 +1,59 @@
+
+/obj/structure/closet/cardboard/relaymove(mob/user, direction)
+	if(opened || move_delay || user.stat || user.stunned || user.weakened || user.paralysis || !isturf(loc) || !has_gravity(loc)/*|| Kawanishi-Noseguchi || Kinunobebashi || Takiyama || Uguisunomori || Tsuzumigataki || Tada || Hirano || Ichinotorii || Uneno || Yamashita || Sasabe || Kofudai || Tokiwadai || Myokenguchi*/)
+		return
```
!

phil235 commented on code/game/objects/structures/crates_lockers/closets/cardboardbox.dm:
```diff
@@ -0,0 +1,59 @@
+
+/mob/living/proc/do_alert_animation(atom/A)
+	var/final_pixel_y = get_standard_pixel_y_offset(lying)
+	..(A, final_pixel_y)
```
there is no parent to call...

duncathan replied:
Why does byond let that compile

phil235 commented on code/game/objects/structures/crates_lockers/closets/cardboardbox.dm:
```diff
@@ -0,0 +1,59 @@
+/mob/living/proc/do_alert_animation(atom/A)
+	var/final_pixel_y = get_standard_pixel_y_offset(lying)
+	..(A, final_pixel_y)
+	floating = 0
```
it's only need after calling animate() not before.

Incoming5643 committed:
Patches up some animation mechanics misconceptions

Incoming5643 commented:
Blame me not being terribly familiar with how animations work. You don't want to know how long it took me just to see that little ! work right.

But yeah calling non existent parents probably isn't required :baby:

Niknakflak commented:
!

KorPhaeron commented on code/game/objects/structures/crates_lockers/closets/cardboardbox.dm:
```diff
@@ -0,0 +1,55 @@
+				alerted << sound('sound/machines/chime.ogg') //HQ HQ!
+				egg_cooldown = 1
+				spawn(3000)
+					egg_cooldown = 0
```
Shouldn't all this egg cooldown be something along the lines of 

if(egg_cooldown < world.time)

And then egg_cooldown = world.time + 3000

Instead of using spawn?

Incoming5643 replied:
But then I'd have to code in midnight rollover protection

duncathan replied:
Is that spess y2k

Iamgoofball replied:
basically yeah

tedward1337 commented:
!

Incoming5643 commented:
I've noticed a very confusing trend where the more lackadaisical and impulsive a pull request I make the more scrutiny it comes under. It's just a booooooooooox.

tedward1337 commented:
More like
Tacti-COOL espionage action am I right guys??

Eddlicious commented:
Could you fold these large cardboard boxes? Y'know, for... Tactical uses.

Razharas commented:
Plz remove silly comments

Incoming5643 commented:
You may pick one to be removed, but whichever you spare must remain enshrined.

PKPenguin321 commented:
fuckin' finally
